core/deployment/src/main/java/io/quarkus/deployment/configuration/RunTimeConfigurationGenerator.java

```diff
@@ -406,11 +406,22 @@ public static void generate(
             // build time defaults
             clinit.writeArrayValue(array, 1, buildTimeRunTimeDefaultValuesConfigSource);
             clinit.invokeVirtualMethod(SRCB_WITH_SOURCES, buildTimeBuilder, array);
-            // add static init sources
-            for (String providerClass : additionalStaticInitConfigSourceProviders) {
-                ResultHandle providerInstance = clinit.newInstance(MethodDescriptor.ofConstructor(providerClass));
-                clinit.invokeStaticMethod(CU_ADD_SOURCE_PROVIDER, buildTimeBuilder, providerInstance);
+            // add safe static sources
```

Comment:
If an reading line `400` correctly, it seems like we are still doing discovery. Did I get that wrong?

Reply:
The method signature have a boolean value to turn the discovery true / false. This call is set to false:

I want to refactor that code one of these days. It is super confusing to deal with the flags and read them on the generator code. I think it would be better if we have a separate method for each `Config` initialization phase so you can see exactly what is going on and if needed change it safely without fear of affecting any of the other modes.

Reply:
Gotcha, thanks!